File: 0713-subarray-product-less-than-k/0713-subarray-product-less-than-k.java
class Solution {
    public int numSubarrayProductLessThanK(int[] nums, int k) {
        int pd=1;
        int c=0;
        int t=0;
      for(int i=0;i<nums.length||t<nums.length;){
          if(i>=nums.length){
              i=t+1;
              t=i;
              pd=1;
              continue;
          }
          pd=pd*nums[i];
          if(pd<k){
            c++;  
          }
          else{
              i=t+1;
              t=i;
              pd=1;
              continue;
          }
          
          i++;
    }
        return c;
}
}

File: 0713-subarray-product-less-than-k/0713-subarray-product-less-than-k.py
class Solution:
    def numSubarrayProductLessThanK(self, nums: list[int], k: int) -> int:
        mul=1
        i=0
        j=0
        ans=0
        while j<len(nums):
            mul=mul*nums[j]
            while(mul>=k and i<=j):
                mul=mul//nums[i]
                i+=1
            ans+=j-i+1
            j+=1
        return ans
        